cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -0,0 +1,1161 @@
+                  RandomAccessIteratorOut,
+                  TransformOp,
+                  PolicyHub>
+{
+  static_assert(::cuda::std::is_same<Offset, ::cuda::std::int32_t>::value
+                  || ::cuda::std::is_same<Offset, ::cuda::std::int64_t>::value,
+                "cub::DeviceTransform is only tested and tuned for 32-bit or 64-bit signed offset types");
+
+  ::cuda::std::tuple<RandomAccessIteratorsIn...> in;
+  RandomAccessIteratorOut out;
+  Offset num_items;
+  TransformOp op;
+  cudaStream_t stream;
+
+  using kernel_ptr_t =
+    decltype(&transform_kernel<typename PolicyHub::max_policy,
+                               Offset,
+                               TransformOp,
+                               RandomAccessIteratorOut,
+                               THRUST_NS_QUALIFIER::try_unwrap_contiguous_iterator_t<RandomAccessIteratorsIn>...>);
+  kernel_ptr_t kernel;
+
+  static constexpr int loaded_bytes_per_iter = loaded_bytes_per_iteration<RandomAccessIteratorsIn...>();
+
+  // TODO(bgruber): I want to write tests for this but those are highly depending on the architecture we are running on?
+  template <typename ActivePolicy, std::size_t... Is>
+  CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE auto configure_memcpy_async_kernel(cuda::std::index_sequence<Is...>)
+    -> PoorExpected<::cuda::std::tuple<THRUST_NS_QUALIFIER::cuda_cub::launcher::triple_chevron, kernel_ptr_t, int>>
+  {
+    // Benchmarking shows that even for a few iteration, this loop takes around 4-7 us, so should not be a concern.
+    using policy_t          = typename ActivePolicy::algo_policy;
+    constexpr int block_dim = policy_t::BLOCK_THREADS;
+    static_assert(block_dim % memcpy_async_alignment == 0,
+                  "BLOCK_THREADS needs to be a multiple of memcpy_async_alignment (16)"); // then tile_size is a
+                                                                                          // multiple of 16-byte
+    const auto max_smem = get_max_shared_memory();
+    if (!max_smem)
+    {
+      return max_smem.error;
+    }
+
+    // TODO(bgruber): we could consider the element size here and ensure that the tilesize is a multiple of 16 bytes.
+    // Increase the number of output elements per thread until we reach the required bytes in flight.
+    int chosen_elem_per_thread = 0;
+    int chosen_tile_size       = 0;
+    int chosen_smem_size       = 0;
+    for (int elem_per_thread = +policy_t::MIN_ITEMS_PER_THREAD; elem_per_thread < +policy_t::MAX_ITEMS_PER_THREAD;
+         ++elem_per_thread)
+    {
+      const auto tile_size = block_dim * elem_per_thread;
+
+      int smem_size   = 0;
+      auto count_smem = [&](int size, int alignment) {
+        smem_size = round_up_to_po2_multiple(smem_size, alignment);
+        // max aligned_base_ptr offset + max padding after == 16
+        smem_size += size * tile_size + ::cuda::std::max(memcpy_async_alignment, memcpy_async_size_multiple);
```

Comment:
Question (double-check): what happens if the data type has an alignment that is larger than the memcpy_async_alignment? Do we need a larger padding in this case?

Reply:
It's quite fun: A type with an alignment of 256, will just report a size of 256, and with a block size of 256 blow the amount of available shared memory on my workstation GPU and fail to come up with a launch configuration.

Reply:
I added tests for this case. In principle, if the type's alignment gets larger than the alignment we require for the copy operation, we just don't need to move the base ptr anywhere and also not introduce any front or tail padding. Only the copy destination in shared memory needs to be appropriately aligned, which affects shared memory consumption.